File: Задание1/prog.swift
// возвращаемое значение функции,
// представляет собой номер (нумерация с 0) операции копирования строки,
// а так же кол-во копируемых символов

func numberCopyOperation(_ s: String) -> Int {
    var i = s.count - (s.count % 2)
    var middle = i / 2

    while i > middle {
        if s[s.index(s.startIndex, offsetBy: i-1)] != s[s.index(s.startIndex, offsetBy: i - middle - 1)] {
            i = i % 2 == 0 ? i : i - 1
            middle = i / 2
        }
        i -= 1
    }

    return middle
}

var s = readLine() ?? ""

let num = numberCopyOperation(s)
let operationsCnt = s.count - num + (num != 0 ? 1 : 0)

print("operations count: \(operationsCnt)", terminator: "")

for i in 0..<operationsCnt {
    let lenOutStr = i + (i < num || num == 0 ? 1 : num)
    if i == 0 {
        print(" = \(String(s.prefix(lenOutStr)))", terminator: "")
    } else {
        print("->\(String(s.prefix(lenOutStr)))", terminator: "")
    }
}
 

File: Задание1/prog.cpp
#include <iostream> 
#include <string>
using namespace std;

// возвращаемое значение функции,
// представляет собой номер (нумерация с 0) операции копирования строки,
// а так же кол-во копируемых символов
size_t number_copy_operation (const string& s)
{
	size_t i = s.size() - (s.size() % 2);
	size_t middle = i / 2;

	for (; i > middle; --i)
	{
		if (s[i-1] != s[i - middle - 1])
		{
			i -= !(i % 2);
			middle = i / 2;
		}
	}

	return middle;
}

int main()
{
	string s;
	cin >> s;
	
	size_t num = number_copy_operation(s);
	size_t operationsCnt = s.size() - num + (num != 0);

	cout << "operations count: " << operationsCnt;

	for (size_t i = 0; i < operationsCnt; ++i)
	{
		size_t lenOutStr = i + (i < num or num == 0 ? 1 : num);

		cout << (i > 0 ? "->" : " = ") << s.substr(0, lenOutStr);
	}
}